src/rum/config.cpp

```diff
@@ -0,0 +1,154 @@
+    char *err_msg =
+        static_cast<char *>(ngx_palloc(cf->pool, sizeof(char) * 256));
+    ngx_snprintf((u_char *)err_msg, 256, "Empty value");
+    return err_msg;
+  }
+
+  (*rum_config)[std::string(key)] = std::string(value);
+  return NGX_CONF_OK;
+}
+
+static std::string make_rum_json_config(
+    std::string_view config_version,
+    const std::unordered_map<std::string, std::string> &config) {
+  rapidjson::Document doc;
+  doc.SetObject();
+
+  rapidjson::Document::AllocatorType &allocator = doc.GetAllocator();
+  doc.AddMember("majorVersion",
+                rapidjson::Value(std::stoi(config_version.data() + 1)),
+                allocator);
+
+  rapidjson::Value rum(rapidjson::kObjectType);
+  for (const auto &[key, value] : config) {
+    if (key == "majorVersion") {
+      continue;
+    } else if (key == "sessionSampleRate" || key == "sessionReplaySampleRate") {
+      rum.AddMember(rapidjson::Value(key.c_str(), allocator).Move(),
+                    rapidjson::Value(std::stod(value)).Move(), allocator);
+    } else if (key == "trackResources" || key == "trackLongTasks" ||
+               key == "trackUserInteractions") {
+      auto b = (value == "true" ? true : false);
+      rum.AddMember(rapidjson::Value(key.c_str(), allocator).Move(),
+                    rapidjson::Value(b).Move(), allocator);
+    } else {
+      rum.AddMember(rapidjson::Value(key.c_str(), allocator).Move(),
+                    rapidjson::Value(value.c_str(), allocator).Move(),
+                    allocator);
+    }
+  }
+
+  doc.AddMember("rum", rum, allocator);
+
+  // Convert the document to a JSON string
+  rapidjson::StringBuffer buffer;
+  rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+  doc.Accept(writer);
+
+  return buffer.GetString();
+}
+
+}  // namespace
+
+extern "C" {
+
+char *on_datadog_rum_config(ngx_conf_t *cf, ngx_command_t *command,
+                            void *conf) {
+  auto *loc_conf = static_cast<datadog::nginx::datadog_loc_conf_t *>(conf);
+
+  auto *values = static_cast<ngx_str_t *>(cf->args->elts);
+
+  std::string_view config_version = datadog::nginx::to_string_view(values[1]);
+  if (!config_version.starts_with('v')) {
+    auto *err_msg =
+        static_cast<char *>(ngx_palloc(cf->pool, sizeof(char) * 256));
+    ngx_snprintf((u_char *)err_msg, 256,
+                 "Invalid version argument provided. Expected version 'v5' but "
+                 "encountered '%s'. Please ensure you are using the correct "
+                 "version format 'v5'",
+                 config_version);
+    return err_msg;
+  }
+
+  std::unordered_map<std::string, std::string> rum_config;
+
+  ngx_conf_t save = *cf;
+  cf->handler = set_config;
+  cf->handler_conf = &rum_config;
+  char *status = ngx_conf_parse(cf, NULL);
+  *cf = save;
+
+  if (status != NGX_CONF_OK) {
+    return status;
+  }
+
+  auto json = make_rum_json_config(config_version, rum_config);
+  if (json.empty()) {
+    auto *err_msg =
```

Comment:
Forgot to return `err_msg` here?

Reply:
Indeed. Good catch.
- [ ] return `err_msg`